card/mod/solid_cache/set/abstract/solid_cache.rb

```diff
@@ -72,21 +77,36 @@ def event_name set, args
   end
 end
 
-def expire_solid_cache
+def expire_solid_cache _changed_card=nil
   return unless solid_cache?
   Auth.as_bot do
     solid_cache_card.delete!
   end
 end
 
-def update_solid_cache
+def update_solid_cache changed_card=nil
   return unless solid_cache?
-  new_content = format(:html)._render_core(solid_cache: false)
+  new_content =
+    if solid_cache_card.new?
+      generate_content_for_cache changed_card
+    else
+      updated_content_for_cache changed_card
+    end
   return unless new_content
   write_to_solid_cache new_content
   new_content
 end
 
+def generate_content_for_cache changed_card=nil
+  format_type = try(:cached_format) || :base
+  format(format_type)._render_core(solid_cache: false,
+                                   changed_card: changed_card)
+end
+
+def updated_content_for_cache _changed_card=nil
+  generate_content_for_cache
+end
```

Comment:
yes, surgical updates using `changed_card` (in WikiRate) instead of regenerating everything